Products/DataCollector/ApplyDataMap.py

```diff
@@ -216,6 +247,9 @@ def _applyDataMapImpl(self, device, datamap):
             self.num_obj_changed,
             logname)
 
+        handle_datamap(device, datamap, self.context)
```

Comment:
Shouldnt we call handle_datamap once we are sure this has successfully completed?https://github.com/zenoss/zenoss-prodbin/blob/004974ca24e7c265a6f704bb59d236f47b2595e6/Products/DataCollector/ApplyDataMap.py#L152-L163 in case the transact fails bc of a conflict error or something?